modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -136,21 +138,43 @@ public PartitionReplicaListener(
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<Object> invoke(ReplicaRequest request) {
-        if (request instanceof ReadWriteSingleRowReplicaRequest) {
-            return processSingleEntryAction((ReadWriteSingleRowReplicaRequest) request);
-        } else if (request instanceof ReadWriteMultiRowReplicaRequest) {
-            return processMultiEntryAction((ReadWriteMultiRowReplicaRequest) request);
-        } else if (request instanceof ReadWriteSwapRowReplicaRequest) {
-            return processTwoEntriesAction((ReadWriteSwapRowReplicaRequest) request);
-        } else if (request instanceof ReadWriteScanRetrieveBatchReplicaRequest) {
-            return processScanRetrieveBatchAction((ReadWriteScanRetrieveBatchReplicaRequest) request);
-        } else if (request instanceof ReadWriteScanCloseReplicaRequest) {
-            processScanCloseAction((ReadWriteScanCloseReplicaRequest) request);
-
-            return CompletableFuture.completedFuture(null);
+        CompletableFuture<Object> fut;
+        if (request instanceof ReadWriteReplicaRequest) {
+            ReadWriteReplicaRequest request0 = (ReadWriteReplicaRequest) request;
+
+            fut = raftClient.refreshAndGetLeaderWithTerm()
+                    .thenCompose((replicaAndTerm) -> {
+                        if (!replicaAndTerm.get2().equals(request0.term())) {
```

Comment:
It's also required to check whether localNode is the same as the current leader.

Reply:
Fixed, but we need to check term only.